Refactor minimap size to be relative to window height

BenoitRoux0 opened the pull request:
Updated the minimap size to be a fraction of the window height, as opposed to a fixed size. This ensures that the minimap maintains its proportion irrespective of window size, improving user experience. The code was also made more readable by changing minimap size from a 2D vector to an integer, which simplified several calculations in minimap.c.

BenoitRoux0 committed:
Refactor minimap size to be relative to window height

Updated the minimap size to be a fraction of the window height, as opposed to a fixed size. This ensures that the minimap maintains its proportion irrespective of window size, improving user experience. The code was also made more readable by changing minimap size from a 2D vector to an integer, which simplified several calculations in minimap.c.

BenoitRoux0 committed:
resolve conflicts

BenoitRoux0 committed:
resolve conflicts

BenoitRoux0 committed:
Merge remote-tracking branch 'origin/bonus-minimap' into bonus-minimap

gd-harco reviewed (changes requested):
Does not compile :
```
➜  cub3D git:(bonus-minimap) ✗ make bonus 
make all IS_BONUS=1
make[1] : on entre dans le répertoire « /home/gd-harco/Documents/42/level_4/cub3D »
cc -Wall -Wextra -Werror  -DWIN_WIDTH=1366 -DWIN_HEIGHT=768 -c bonus/srcs/main_bonus.c -o bonus/srcs/main_bonus.o -Ilibft/includes -Iminilibx-linux -Ibonus/incs
make[1]: ***  Aucune règle pour fabriquer la cible « bonus/srcs/minimap.o », nécessaire pour « cub3D_bonus ». Arrêt.
make[1] : on quitte le répertoire « /home/gd-harco/Documents/42/level_4/cub3D »
make: *** [Makefile:84 : bonus] Erreur 2
```

BenoitRoux0 committed:
Update Makefile and refactor minimapBonus.c for bonus version

Updated the Makefile to use minimap_bonus.c instead of minimap.c when building the bonus version of the program. This was necessary to separate logic for the bonus and standard versions of the application.

Refactored minimap_bonus.c to fix a bug in the rendering of the minimap where the player's angle was not correctly converted from radians to degrees, resulting in incorrectly displayed player orientation on the minimap.

BenoitRoux0 committed:
Merge remote-tracking branch 'origin/master' into bonus-minimap

BenoitRoux0 committed:
Update Makefile for bonus version and fix minimap rendering bug

Updated Makefile to include 'init_angles_bonus.c' for bonus version to properly separate application versions. This change improves separation of functionalities between standard and bonus version. Additionally, fixed a bug in 'minimap_bonus.c' where the player's angle was incorrectly converted from radians to degrees, leading to a disparity in player orientation on the minimap.

BenoitRoux0 reviewed (commented):
all fixed

gd-harco commented on Makefile:
le makefile ici

gd-harco commented on Makefile:
```diff
@@ -36,17 +36,17 @@ endif
 MLX =		minilibx-linux/libmlx.a
 LIBFT =		libft/libft.a
 
 ifeq ($(shell command -v xrandr 2> /dev/null),)
```
ici le merge a merde

gd-harco reviewed (approved):
Makefile ne prend pas en compte le changement de résolution de la fenêtre, je fais une PR pour fix
